aptible/resource_endpoint.go

```diff
@@ -113,6 +124,27 @@ func resourceEndpoint() *schema.Resource {
 	}
 }
 
+func resourceEndpointValidate(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	d := ResourceDiff{ResourceDiff: diff}
+	interfaceContainerPortsSlice := d.Get("container_ports").([]interface{})
+	containerPorts, _ := aptible.MakeInt64Slice(interfaceContainerPortsSlice)
+	containerPort, _ := (d.Get("container_port").(int))
+	endpointType := d.Get("endpoint_type").(string)
+	var err error
+
+	// container_port and container ports are mutually exclusive fields
+	if containerPort != 0 && len(containerPorts) != 0 {
+		err = multierror.Append(err, fmt.Errorf("do not specify container ports AND container port (see terraform docs)"))
+	}
+
+	// container ports can only be used with tls/tcp
```

Comment:
Isn't the inverse also be true? `container_port` can only be used with `https`?

Reply:
Confirmed that TCP/TLS Endpoints don't support `container_port`

Reply:
it is, i'm adding that validation in after chatting and verifying w/ you (confirming what michael sent to me earlier)

Reply:
this should now be addressed